app/controllers/teams/users_controller.rb

```diff
@@ -1,5 +1,33 @@
 class Teams::UsersController < AuthenticatedController
-   before_action -> { authorize team, policy_class: TeamUsersPolicy }
+    before_action -> { authorize team, policy_class: TeamUsersPolicy }
+
+    def new
+      @user = User.new
+    end
+  
+    def create
+      add_email = user_params.require(:email).downcase
+      if not valid_email_address?(add_email)
+        @user = User.new
+        flash[:error] = I18n.t('teams.users.create.invalid_email', email: add_email)
+        render :new
+      else
+        existing_user_emails = team.users.map(&:email)
+        if existing_user_emails.include?(add_email)
+          @user = User.new
+          flash[:error] = I18n.t('teams.users.create.already_member', email: add_email)
+          render :new
+        else
+          @user = User.where(email: add_email).first || User.new(email: add_email)
+          if team.update(users: team.users + [@user])
+              flash[:success] = I18n.t('teams.users.create.success', email: add_email)
+              redirect_to team_path(team.id)
```

Comment:
`redirect_to team_path(team.id) and return`, if you move all of the `render :new`'s get moved down